### PR TITLE
Add Claims service

### PR DESCRIPTION
**Pull request by jchen293**
# Description

- Add new `Claim` service for filing claims on EasyPost shipments and insurances
# Testing

Add unit tests

# Pull Request Type

Please select the option(s) that are relevant to this PR.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Improvement (fixing a typo, updating readme, renaming a variable name, etc)

### CODE REVIEW

File: src/main/java/com/easypost/model/DateDeserializer.java [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+import com.google.gson.JsonParseException;
+
+import java.lang.reflect.Type;
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.util.Date;
+import java.util.TimeZone;
+
+public class DateDeserializer implements JsonDeserializer<Date> {
+    private static final String[] DATE_FORMATS = new String[]{
```
Comment (nwithan8) [inline src/main/java/com/easypost/model/DateDeserializer.java:15]: Sigh...

File: src/main/java/com/easypost/model/ClaimCollection.java [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+
+    @Override
+    protected Map<String, Object> buildNextPageParameters(List<Claim> claims, Integer pageSize)
+            throws EndOfPaginationError {
+        String lastId = claims.get(claims.size() - 1).getId();
+
+        Map<String, Object> parameters = new java.util.HashMap<>();
+        parameters.put("before_id", lastId);
+
+        if (pageSize != null) {
```
Comment (nwithan8) [inline src/main/java/com/easypost/model/ClaimCollection.java:21]: Need to re-inject the "status" and "type" filters used on the original request (see Address or Report for example)

File: src/main/java/com/easypost/model/DateDeserializer.java (hunk 1 above)
Comment (jchen293) [reply src/main/java/com/easypost/model/DateDeserializer.java:15]: Claims API endpoints return a different date format ("yyyy-MM-dd'T'HH:mm:ss") and we can't deserialize it. We need to a custom deserializer

File: src/main/java/com/easypost/model/ClaimCollection.java (hunk 2 above)
Comment (jchen293) [reply src/main/java/com/easypost/model/ClaimCollection.java:28]: added

**Review by nwithan8 — APPROVED** [review]

File: src/main/java/com/easypost/service/ClaimService.java [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+     * Get a list of Claims.
+     *
+     * @param params a map of parameters
+     * @return ClaimCollection object
+     * @throws EasyPostException when the request fails.
+     */
+    public ClaimCollection all(final Map<String, Object> params) throws EasyPostException {
+        String endpoint = "claims";
+
+        return Requestor.request(RequestMethod.GET, endpoint, params, ClaimCollection.class, client);
```
Comment (nwithan8) [inline src/main/java/com/easypost/service/ClaimService.java:63]: "status" and "type" need to be extracted and stored in the resultant collection.

**Review by nwithan8 — CHANGES_REQUESTED** [review]

File: src/main/java/com/easypost/service/ClaimService.java (hunk 3 above)
Comment (jchen293) [reply src/main/java/com/easypost/service/ClaimService.java]: Good catch! Added

**Review by nwithan8 — APPROVED** [review]

**Pull request merged**